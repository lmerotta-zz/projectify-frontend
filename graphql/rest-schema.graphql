type LoginInput {
  username: String!
  password: String!
}

type NoResponse {
  NoResponse: String
}

extend type Mutation {
  login(input: LoginInput): NoResponse
}

directive @rest(
  type: String!
  path: String!
  method: String!
  endpoint: String!
  bodyKey: String!
) on FIELD
