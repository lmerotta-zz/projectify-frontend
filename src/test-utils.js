import React from "react";
import { render } from "@testing-library/react";
import { CacheProvider } from "@emotion/react";
import createCache from "@emotion/cache";

const cache = createCache({ key: "iodajdoiwanoawdni" });
cache.compat = true;

const AllTheProviders = ({ children }) => {
  return <CacheProvider value={cache}>{children}</CacheProvider>;
};

const customRender = (ui, options) =>
  render(ui, { wrapper: AllTheProviders, ...options });

// re-export everything
export * from "@testing-library/react";

// override render method
export { customRender as render };
