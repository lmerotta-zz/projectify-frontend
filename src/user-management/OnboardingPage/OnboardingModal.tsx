/** @jsxImportSource @emotion/react */
import "twin.macro";
import {
  Button,
  FileInput,
  FormGroup,
  Modal,
  ModalBody,
  ModalFooter,
  ModalHeader,
} from "components";
import gql from "graphql-tag";
import { useMutation, useQuery } from "@apollo/client";
import { getCurrentUser } from "apollo/types/getCurrentUser";
import * as yup from "yup";
import { FormProvider, useForm } from "react-hook-form";
import { yupResolver } from "@hookform/resolvers/yup";
import { Input } from "components";
import { onboardVariables, onboard_onboardUser } from "apollo/types/onboard";

export const CURRENT_USER_QUERY = gql`
  query getCurrentUser {
    currentUser {
      firstName
      lastName
    }
  }
`;

export const ONBOARD_USER_MUTATION = gql`
  mutation onboard($firstName: String!, $lastName: String!, $picture: Upload!) {
    onboardUser(
      input: { picture: $picture, firstName: $firstName, lastName: $lastName }
    ) {
      clientMutationId
    }
  }
`;

type OnboardingFormType = {
  picture: File;
  firstName: string;
  lastName: string;
};

const schema = yup.object().shape({
  picture: yup.mixed().required(),
  firstName: yup.string().required(),
  lastName: yup.string().required(),
});

const OnboardingModal = () => {
  const currentUserQuery = useQuery<getCurrentUser>(CURRENT_USER_QUERY);
  const [onboardUser] = useMutation<onboard_onboardUser, onboardVariables>(
    ONBOARD_USER_MUTATION
  );

  const form = useForm<OnboardingFormType>({
    resolver: yupResolver(schema),
  });

  const onSubmit = form.handleSubmit(async (data) => {
    await onboardUser({
      variables: data,
    });
  });

  return currentUserQuery.data && currentUserQuery.data.currentUser ? (
    <Modal isOpen>
      <ModalHeader>Welcome to projectify!</ModalHeader>
      <FormProvider {...form}>
        <form onSubmit={onSubmit}>
          <ModalBody>
            <h4 tw="text-base text-dark mb-4">
              Before continuing, please confirm that these informations are
              valid
            </h4>
            <FormGroup>
              <Input
                type="text"
                name="firstName"
                label="firstName"
                defaultValue={currentUserQuery.data.currentUser.firstName}
                ref={form.register}
              />
            </FormGroup>
            <FormGroup>
              <Input
                type="text"
                name="lastName"
                label="lastName"
                ref={form.register}
                defaultValue={currentUserQuery.data.currentUser.lastName}
              />
            </FormGroup>
            <FormGroup tw="w-full justify-center">
              <FileInput
                name="picture"
                label="Profile picture"
                dropzone={{
                  accept: ["image/*"],
                  maxFiles: 1,
                }}
              />
            </FormGroup>
          </ModalBody>
          <ModalFooter>
            <Button type="submit">Submit</Button>
          </ModalFooter>
        </form>
      </FormProvider>
    </Modal>
  ) : null;
};

export default OnboardingModal;
