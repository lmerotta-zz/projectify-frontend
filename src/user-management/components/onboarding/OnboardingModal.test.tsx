import { render, fireEvent, act } from "test-utils";
import { MockedProvider } from "@apollo/client/testing";
import OnboardingModal, { ONBOARD_USER_MUTATION } from "./OnboardingModal";
import { toast } from "react-toastify";
import { GraphQLError } from "graphql";
import { UserStatus } from "generated/graphql";

describe("OnboardingModal unit tests", () => {
  beforeEach(() => {
    window.URL.createObjectURL = jest.fn().mockImplementation(() => "url");
  });
  it("throws an error if the user is already onboarded", async () => {
    const toastSpy = jest.spyOn(toast, "error");
    const file = new File(["file"], "test.png", {
      type: "image/png",
    });

    const mocks = [
      {
        request: {
          query: ONBOARD_USER_MUTATION,
          variables: {
            firstName: "test",
            lastName: "tost",
            picture: file,
          },
        },
        result: {
          errors: [
            new GraphQLError(
              "onboarding failed",
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              {
                exception_code: 4203,
              }
            ),
          ],
        },
      },
    ];

    const result = render(
      <MockedProvider mocks={mocks}>
        <OnboardingModal
          user={{
            __typename: "User",
            firstName: "test",
            lastName: "tost",
            id: "abcd",
            status: UserStatus.SignedUpOauth,
          }}
        />
      </MockedProvider>
    );

    expect(result.getByTestId("input-firstName")).not.toHaveAttribute(
      "readonly"
    );
    expect(result.getByTestId("input-lastName")).not.toHaveAttribute(
      "readonly"
    );

    const fileInput = result.getByTestId("file-input-picture");
    await act(async () => {
      Object.defineProperty(fileInput, "files", {
        value: [file],
      });
    });

    await act(async () => {
      await fireEvent.drop(fileInput);
    });

    await act(async () => {
      await fireEvent.click(result.getByTestId("onboarding-modal-submit"));
    });

    await act(async () => {
      await new Promise((resolve) => setTimeout(resolve, 1000));
    });

    expect(toastSpy).toHaveBeenCalledWith(
      "usermanagement.onboarding.modal.error.user_already_onboarded"
    );
    expect(result.asFragment()).toMatchSnapshot();
  });

  it("throws an error if the server fails", async () => {
    const toastSpy = jest.spyOn(toast, "error");
    const file = new File(["file"], "test.png", {
      type: "image/png",
    });

    const mocks = [
      {
        request: {
          query: ONBOARD_USER_MUTATION,
          variables: {
            firstName: "test",
            lastName: "tost",
            picture: file,
          },
        },
        result: {
          errors: [new GraphQLError("onboarding failed")],
        },
      },
    ];

    const result = render(
      <MockedProvider mocks={mocks}>
        <OnboardingModal
          user={{
            __typename: "User",
            firstName: "test",
            lastName: "tost",
            id: "abcd",
            status: UserStatus.SignedUpOauth,
          }}
        />
      </MockedProvider>
    );

    const fileInput = result.getByTestId("file-input-picture");
    await act(async () => {
      Object.defineProperty(fileInput, "files", {
        value: [file],
      });
    });

    await act(async () => {
      await fireEvent.drop(fileInput);
    });

    await act(async () => {
      await fireEvent.click(result.getByTestId("onboarding-modal-submit"));
    });

    await act(async () => {
      await new Promise((resolve) => setTimeout(resolve, 0));
    });

    expect(toastSpy).toHaveBeenCalledWith(
      "global.errors.internal-server-error"
    );
  });

  it("Shows a success message if onboarding is successfull", async () => {
    const toastSpy = jest.spyOn(toast, "success");
    const file = new File(["file"], "test.png", {
      type: "image/png",
    });

    const mocks = [
      {
        request: {
          query: ONBOARD_USER_MUTATION,
          variables: {
            firstName: "test",
            lastName: "tost",
            picture: file,
          },
        },
        result: {
          data: {
            user: {
              id: "abcd",
              status: UserStatus.Onboarded,
              profilePictureUrl: "url",
            },
          },
        },
      },
    ];

    const result = render(
      <MockedProvider mocks={mocks}>
        <OnboardingModal
          user={{
            __typename: "User",
            firstName: "test",
            lastName: "tost",
            id: "abcd",
            status: UserStatus.SignedUpOauth,
          }}
        />
      </MockedProvider>
    );

    const fileInput = result.getByTestId("file-input-picture");
    await act(async () => {
      Object.defineProperty(fileInput, "files", {
        value: [file],
      });
    });

    await act(async () => {
      await fireEvent.drop(fileInput);
    });

    await act(async () => {
      await fireEvent.click(result.getByTestId("onboarding-modal-submit"));
    });

    await act(async () => {
      await new Promise((resolve) => setTimeout(resolve, 0));
    });

    expect(toastSpy).toHaveBeenCalledWith(
      "usermanagement.onboarding.modal.user_onboarded"
    );
  });

  it("Puts first name and last name as readonly if user is signed up normally", async () => {
    const result = render(
      <MockedProvider mocks={[]}>
        <OnboardingModal
          user={{
            __typename: "User",
            firstName: "test",
            lastName: "tost",
            id: "abcd",
            status: UserStatus.SignedUp,
          }}
        />
      </MockedProvider>
    );

    expect(result.getByTestId("input-firstName")).toHaveAttribute("readonly");
    expect(result.getByTestId("input-lastName")).toHaveAttribute("readonly");
    expect(result.asFragment()).toMatchSnapshot();
  });
});
