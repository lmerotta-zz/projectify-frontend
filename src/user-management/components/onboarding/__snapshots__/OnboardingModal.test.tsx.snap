// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OnboardingModal unit tests Puts first name and last name as readonly if user is signed up normally 1`] = `
<DocumentFragment>
  .emotion-0 {
  position: fixed;
  width: 100%;
  height: 100%;
  --tw-bg-opacity: 1;
  background-color: rgba(147, 51, 234, var(--tw-bg-opacity));
  padding: 1rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  top: 0px;
  left: 0px;
  bottom: 0px;
  right: 0px;
  z-index: 1050;
}

.emotion-1 {
  border-radius: 0.375rem;
  --tw-bg-opacity: 1;
  background-color: rgba(255, 255, 255, var(--tw-bg-opacity));
  --tw-shadow: 0 25px 50px -12px rgba(0, 0, 0, 0.25);
  box-shadow: var(--tw-ring-offset-shadow, 0 0 #0000),var(--tw-ring-shadow, 0 0 #0000),var(--tw-shadow);
  width: 100%;
  height: auto;
  position: relative;
  z-index: 10;
}

@media (min-width: 768px) {
  .emotion-1 {
    width: 80%;
  }
}

@media (min-width: 1024px) {
  .emotion-1 {
    width: 50%;
  }
}

@media (min-width: 1280px) {
  .emotion-1 {
    width: 40%;
  }
}

.emotion-2 {
  padding: 1rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  justify-content: space-between;
  border-bottom-width: 2px;
  --tw-border-opacity: 1;
  border-color: rgba(229, 231, 235, var(--tw-border-opacity));
}

.emotion-3 {
  font-size: 1.125rem;
  line-height: 1.75rem;
  font-weight: 100;
  --tw-text-opacity: 1;
  color: rgba(39, 39, 42, var(--tw-text-opacity));
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
}

.emotion-4 {
  padding-left: 1rem;
  padding-right: 1rem;
  padding-top: 1rem;
  padding-bottom: 1rem;
}

.emotion-5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  margin-left: -0.5rem;
  margin-right: -0.5rem;
  -webkit-box-flex-wrap: wrap;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
}

.emotion-6 {
  padding-left: 0.5rem;
  padding-right: 0.5rem;
  margin-bottom: 1rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  max-width: 100%;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
}

@media (min-width: 768px) {
  .emotion-6 {
    -webkit-flex-basis: 0;
    -ms-flex-preferred-size: 0;
    flex-basis: 0;
  }
}

.emotion-7 {
  font-size: 1rem;
  line-height: 1.5rem;
  --tw-text-opacity: 1;
  color: rgba(39, 39, 42, var(--tw-text-opacity));
}

.emotion-10 {
  display: block;
  font-weight: 500;
  margin-bottom: 0.25rem;
}

.emotion-11 {
  outline: 2px solid transparent;
  outline-offset: 2px;
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: stretch;
  -webkit-box-align: stretch;
  -ms-flex-align: stretch;
  align-items: stretch;
}

.emotion-12 {
  width: 100%;
  position: relative;
}

.emotion-13 {
  width: 100%;
  padding: 0.5rem;
  font-size: 1rem;
  line-height: 1.5rem;
  border-width: 1px;
  --tw-border-opacity: 1;
  border-color: rgba(156, 163, 175, var(--tw-border-opacity));
  --tw-bg-opacity: 1;
  background-color: rgba(255, 255, 255, var(--tw-bg-opacity));
  -webkit-appearance: none;
  -moz-appearance: none;
  -ms-appearance: none;
  appearance: none;
  outline: 2px solid transparent;
  outline-offset: 2px;
  transition-property: background-color,border-color,color,fill,stroke;
  transition-timing-function: cubic-bezier(0.4, 0, 0.2, 1);
  transition-duration: 150ms;
  --tw-bg-opacity: 1;
  background-color: rgba(229, 231, 235, var(--tw-bg-opacity));
}

.emotion-13:focus {
  --tw-border-opacity: 1;
  border-color: rgba(147, 51, 234, var(--tw-border-opacity));
}

.emotion-13:focus {
  --tw-border-opacity: 1;
  border-color: rgba(156, 163, 175, var(--tw-border-opacity));
}

.emotion-22 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.emotion-23 {
  width: 66.666667%;
  margin-right: 1rem;
}

.emotion-24 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  border-width: 1px;
  --tw-border-opacity: 1;
  border-color: rgba(147, 51, 234, var(--tw-border-opacity));
  border-radius: 0.375rem;
  padding-top: 0.25rem;
  padding-left: 0.75rem;
  padding-right: 0.75rem;
  --tw-bg-opacity: 1;
  background-color: rgba(245, 243, 255, var(--tw-bg-opacity));
}

.emotion-25 {
  width: 100%;
  cursor: pointer;
  height: 8rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-25:focus {
  --tw-ring-offset-shadow: var(--tw-ring-inset) 0 0 0 var(--tw-ring-offset-width) var(--tw-ring-offset-color);
  --tw-ring-shadow: var(--tw-ring-inset) 0 0 0 calc(0px + var(--tw-ring-offset-width)) var(--tw-ring-color);
  box-shadow: var(--tw-ring-offset-shadow),var(--tw-ring-shadow),var(--tw-shadow, 0 0 #0000);
  outline: 2px solid transparent;
  outline-offset: 2px;
}

.emotion-26 {
  --tw-text-opacity: 1;
  color: rgba(147, 51, 234, var(--tw-text-opacity));
}

.emotion-27 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin-top: 0.5rem;
  -webkit-box-flex: 0;
  -webkit-flex-grow: 0;
  -ms-flex-positive: 0;
  flex-grow: 0;
  width: 33.333333%;
}

.emotion-28 {
  font-size: 0.875rem;
  line-height: 1.25rem;
}

.emotion-29 {
  padding: 1rem;
  border-top-width: 2px;
  --tw-border-opacity: 1;
  border-color: rgba(229, 231, 235, var(--tw-border-opacity));
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: end;
  -ms-flex-pack: end;
  -webkit-justify-content: flex-end;
  justify-content: flex-end;
}

.emotion-30 {
  transition-property: background-color,border-color,color,fill,stroke,opacity,box-shadow,transform;
  transition-timing-function: cubic-bezier(0.4, 0, 0.2, 1);
  transition-duration: 150ms;
  padding-left: 0.75rem;
  padding-right: 0.75rem;
  padding-top: 0.75rem;
  padding-bottom: 0.75rem;
  --tw-text-opacity: 1;
  color: rgba(255, 255, 255, var(--tw-text-opacity));
  --tw-bg-opacity: 1;
  background-color: rgba(147, 51, 234, var(--tw-bg-opacity));
}

.emotion-30:focus {
  outline: 2px solid transparent;
  outline-offset: 2px;
}

.emotion-30:hover {
  --tw-bg-opacity: 1;
  background-color: rgba(126, 34, 206, var(--tw-bg-opacity));
}

<div
    class="emotion-0"
    data-testid="modal-backdrop"
    style="opacity: 0;"
  >
    <div
      class="emotion-1"
      style="opacity: 0; transform: translateY(-100%) translateZ(0);"
    >
      <div
        class="emotion-2"
      >
        <h5
          class="emotion-3"
        >
          usermanagement.onboarding.modal.title
        </h5>
      </div>
      <form>
        <div
          class="emotion-4"
        >
          <div
            class="emotion-5"
          >
            <div
              class="emotion-6"
            >
              <h4
                class="emotion-7"
              >
                usermanagement.onboarding.modal.introduction_text
              </h4>
            </div>
          </div>
          <div
            class="emotion-5"
          >
            <div
              class="emotion-6"
            >
              <label
                class="emotion-10"
              >
                usermanagement.onboarding.modal.form.label_firstName
              </label>
              <div
                class="emotion-11"
              >
                <div
                  class="emotion-12"
                >
                  <input
                    class="emotion-13"
                    data-testid="input-firstName"
                    name="firstName"
                    placeholder=" "
                    readonly=""
                    type="text"
                    value="test"
                  />
                </div>
              </div>
            </div>
            <div
              class="emotion-6"
            >
              <label
                class="emotion-10"
              >
                usermanagement.onboarding.modal.form.label_lastName
              </label>
              <div
                class="emotion-11"
              >
                <div
                  class="emotion-12"
                >
                  <input
                    class="emotion-13"
                    data-testid="input-lastName"
                    name="lastName"
                    placeholder=" "
                    readonly=""
                    type="text"
                    value="tost"
                  />
                </div>
              </div>
            </div>
          </div>
          <div
            class="emotion-5"
          >
            <div
              class="emotion-6"
            >
              <label
                class="emotion-10"
              >
                usermanagement.onboarding.modal.form.label_picture
              </label>
              <div
                class="emotion-22"
              >
                <div
                  class="emotion-23"
                >
                  <div
                    class="emotion-24"
                  >
                    <div
                      class="emotion-25"
                      data-testid="file-input-picture"
                      tabindex="0"
                    >
                      <p
                        class="emotion-26"
                      >
                        components.form.fileinput.help_message
                      </p>
                      <input
                        accept="image/jpeg,image/png"
                        autocomplete="off"
                        multiple=""
                        name="picture"
                        style="display: none;"
                        tabindex="-1"
                        type="file"
                      />
                    </div>
                  </div>
                </div>
                <div
                  class="emotion-27"
                >
                  <p
                    class="emotion-28"
                  >
                    usermanagement.onboarding.modal.form.picture_preview
                  </p>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="emotion-29"
        >
          <button
            class="emotion-30"
            data-testid="onboarding-modal-submit"
            type="submit"
          >
            usermanagement.onboarding.modal.form.btn_submit
          </button>
        </div>
      </form>
    </div>
  </div>
</DocumentFragment>
`;

exports[`OnboardingModal unit tests throws an error if the user is already onboarded 1`] = `
<DocumentFragment>
  .emotion-0 {
  position: fixed;
  width: 100%;
  height: 100%;
  --tw-bg-opacity: 1;
  background-color: rgba(147, 51, 234, var(--tw-bg-opacity));
  padding: 1rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  top: 0px;
  left: 0px;
  bottom: 0px;
  right: 0px;
  z-index: 1050;
}

.emotion-1 {
  border-radius: 0.375rem;
  --tw-bg-opacity: 1;
  background-color: rgba(255, 255, 255, var(--tw-bg-opacity));
  --tw-shadow: 0 25px 50px -12px rgba(0, 0, 0, 0.25);
  box-shadow: var(--tw-ring-offset-shadow, 0 0 #0000),var(--tw-ring-shadow, 0 0 #0000),var(--tw-shadow);
  width: 100%;
  height: auto;
  position: relative;
  z-index: 10;
}

@media (min-width: 768px) {
  .emotion-1 {
    width: 80%;
  }
}

@media (min-width: 1024px) {
  .emotion-1 {
    width: 50%;
  }
}

@media (min-width: 1280px) {
  .emotion-1 {
    width: 40%;
  }
}

.emotion-2 {
  padding: 1rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  justify-content: space-between;
  border-bottom-width: 2px;
  --tw-border-opacity: 1;
  border-color: rgba(229, 231, 235, var(--tw-border-opacity));
}

.emotion-3 {
  font-size: 1.125rem;
  line-height: 1.75rem;
  font-weight: 100;
  --tw-text-opacity: 1;
  color: rgba(39, 39, 42, var(--tw-text-opacity));
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
}

.emotion-4 {
  padding-left: 1rem;
  padding-right: 1rem;
  padding-top: 1rem;
  padding-bottom: 1rem;
}

.emotion-5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  margin-left: -0.5rem;
  margin-right: -0.5rem;
  -webkit-box-flex-wrap: wrap;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
}

.emotion-6 {
  padding-left: 0.5rem;
  padding-right: 0.5rem;
  margin-bottom: 1rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  max-width: 100%;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
}

@media (min-width: 768px) {
  .emotion-6 {
    -webkit-flex-basis: 0;
    -ms-flex-preferred-size: 0;
    flex-basis: 0;
  }
}

.emotion-7 {
  font-size: 1rem;
  line-height: 1.5rem;
  --tw-text-opacity: 1;
  color: rgba(39, 39, 42, var(--tw-text-opacity));
}

.emotion-10 {
  display: block;
  font-weight: 500;
  margin-bottom: 0.25rem;
}

.emotion-11 {
  outline: 2px solid transparent;
  outline-offset: 2px;
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: stretch;
  -webkit-box-align: stretch;
  -ms-flex-align: stretch;
  align-items: stretch;
}

.emotion-12 {
  width: 100%;
  position: relative;
}

.emotion-13 {
  width: 100%;
  padding: 0.5rem;
  font-size: 1rem;
  line-height: 1.5rem;
  border-width: 1px;
  --tw-border-opacity: 1;
  border-color: rgba(156, 163, 175, var(--tw-border-opacity));
  --tw-bg-opacity: 1;
  background-color: rgba(255, 255, 255, var(--tw-bg-opacity));
  -webkit-appearance: none;
  -moz-appearance: none;
  -ms-appearance: none;
  appearance: none;
  outline: 2px solid transparent;
  outline-offset: 2px;
  transition-property: background-color,border-color,color,fill,stroke;
  transition-timing-function: cubic-bezier(0.4, 0, 0.2, 1);
  transition-duration: 150ms;
}

.emotion-13:focus {
  --tw-border-opacity: 1;
  border-color: rgba(147, 51, 234, var(--tw-border-opacity));
}

.emotion-22 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.emotion-23 {
  width: 66.666667%;
  margin-right: 1rem;
}

.emotion-24 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  border-width: 1px;
  --tw-border-opacity: 1;
  border-color: rgba(147, 51, 234, var(--tw-border-opacity));
  border-radius: 0.375rem;
  padding-top: 0.25rem;
  padding-left: 0.75rem;
  padding-right: 0.75rem;
  --tw-bg-opacity: 1;
  background-color: rgba(245, 243, 255, var(--tw-bg-opacity));
}

.emotion-25 {
  width: 100%;
  cursor: pointer;
  height: 8rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-25:focus {
  --tw-ring-offset-shadow: var(--tw-ring-inset) 0 0 0 var(--tw-ring-offset-width) var(--tw-ring-offset-color);
  --tw-ring-shadow: var(--tw-ring-inset) 0 0 0 calc(0px + var(--tw-ring-offset-width)) var(--tw-ring-color);
  box-shadow: var(--tw-ring-offset-shadow),var(--tw-ring-shadow),var(--tw-shadow, 0 0 #0000);
  outline: 2px solid transparent;
  outline-offset: 2px;
}

.emotion-26 {
  --tw-text-opacity: 1;
  color: rgba(147, 51, 234, var(--tw-text-opacity));
}

.emotion-27 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin-top: 0.5rem;
  -webkit-box-flex: 0;
  -webkit-flex-grow: 0;
  -ms-flex-positive: 0;
  flex-grow: 0;
  width: 33.333333%;
}

.emotion-28 {
  font-size: 0.875rem;
  line-height: 1.25rem;
}

.emotion-29 {
  height: 5rem;
  width: 5rem;
  margin-top: 0.5rem;
  border-radius: 9999px;
}

.emotion-30 {
  padding: 1rem;
  border-top-width: 2px;
  --tw-border-opacity: 1;
  border-color: rgba(229, 231, 235, var(--tw-border-opacity));
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: end;
  -ms-flex-pack: end;
  -webkit-justify-content: flex-end;
  justify-content: flex-end;
}

.emotion-31 {
  transition-property: background-color,border-color,color,fill,stroke,opacity,box-shadow,transform;
  transition-timing-function: cubic-bezier(0.4, 0, 0.2, 1);
  transition-duration: 150ms;
  padding-left: 0.75rem;
  padding-right: 0.75rem;
  padding-top: 0.75rem;
  padding-bottom: 0.75rem;
  --tw-text-opacity: 1;
  color: rgba(255, 255, 255, var(--tw-text-opacity));
  --tw-bg-opacity: 1;
  background-color: rgba(147, 51, 234, var(--tw-bg-opacity));
}

.emotion-31:focus {
  outline: 2px solid transparent;
  outline-offset: 2px;
}

.emotion-31:hover {
  --tw-bg-opacity: 1;
  background-color: rgba(126, 34, 206, var(--tw-bg-opacity));
}

<div
    class="emotion-0"
    data-testid="modal-backdrop"
    style="opacity: 1;"
  >
    <div
      class="emotion-1"
      style="opacity: 1; transform: translateY(0%) translateZ(0);"
    >
      <div
        class="emotion-2"
      >
        <h5
          class="emotion-3"
        >
          usermanagement.onboarding.modal.title
        </h5>
      </div>
      <form>
        <div
          class="emotion-4"
        >
          <div
            class="emotion-5"
          >
            <div
              class="emotion-6"
            >
              <h4
                class="emotion-7"
              >
                usermanagement.onboarding.modal.introduction_text
              </h4>
            </div>
          </div>
          <div
            class="emotion-5"
          >
            <div
              class="emotion-6"
            >
              <label
                class="emotion-10"
              >
                usermanagement.onboarding.modal.form.label_firstName
              </label>
              <div
                class="emotion-11"
                style="transform: translate3d(0px, 0px, 0) scale(NaN, NaN); transform-origin: 50% 50% 0;"
              >
                <div
                  class="emotion-12"
                  style="transform: translate3d(NaNpx, NaNpx, 0) scale(NaN, NaN); transform-origin: 50% 50% 0;"
                >
                  <input
                    class="emotion-13"
                    data-testid="input-firstName"
                    name="firstName"
                    placeholder=" "
                    type="text"
                    value="test"
                  />
                </div>
              </div>
            </div>
            <div
              class="emotion-6"
            >
              <label
                class="emotion-10"
              >
                usermanagement.onboarding.modal.form.label_lastName
              </label>
              <div
                class="emotion-11"
                style="transform: translate3d(0px, 0px, 0) scale(NaN, NaN); transform-origin: 50% 50% 0;"
              >
                <div
                  class="emotion-12"
                  style="transform: translate3d(NaNpx, NaNpx, 0) scale(NaN, NaN); transform-origin: 50% 50% 0;"
                >
                  <input
                    class="emotion-13"
                    data-testid="input-lastName"
                    name="lastName"
                    placeholder=" "
                    type="text"
                    value="tost"
                  />
                </div>
              </div>
            </div>
          </div>
          <div
            class="emotion-5"
          >
            <div
              class="emotion-6"
            >
              <label
                class="emotion-10"
              >
                usermanagement.onboarding.modal.form.label_picture
              </label>
              <div
                class="emotion-22"
              >
                <div
                  class="emotion-23"
                >
                  <div
                    class="emotion-24"
                  >
                    <div
                      class="emotion-25"
                      data-testid="file-input-picture"
                      tabindex="0"
                    >
                      <p
                        class="emotion-26"
                      >
                        components.form.fileinput.help_message
                      </p>
                      <input
                        accept="image/jpeg,image/png"
                        autocomplete="off"
                        multiple=""
                        name="picture"
                        style="display: none;"
                        tabindex="-1"
                        type="file"
                      />
                    </div>
                  </div>
                </div>
                <div
                  class="emotion-27"
                >
                  <p
                    class="emotion-28"
                  >
                    usermanagement.onboarding.modal.form.picture_preview
                  </p>
                  <img
                    alt="profile-preview"
                    class="emotion-29"
                    src="url"
                  />
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="emotion-30"
        >
          <button
            class="emotion-31"
            data-testid="onboarding-modal-submit"
            type="submit"
          >
            usermanagement.onboarding.modal.form.btn_submit
          </button>
        </div>
      </form>
    </div>
  </div>
</DocumentFragment>
`;
